libs/cookies/src/lib/directives/has-cookie/has-cookie.directive.ts

```diff
@@ -0,0 +1,175 @@
+import {
+	ChangeDetectorRef,
+	ComponentRef,
+	Directive,
+	EmbeddedViewRef,
+	Inject,
+	Input,
+	Optional,
+	TemplateRef,
+	Type,
+	ViewContainerRef,
+} from '@angular/core';
+import { Subject, tap, takeUntil, combineLatest, map } from 'rxjs';
+import { flatten } from 'lodash';
+
+import { NgxHasCookieConfiguration } from '../../types';
+import { NgxCookieService } from '../../services';
+import { NgxCookiesFallBackComponent } from '../../abstracts';
+import { NgxCookiesFallbackComponentToken } from '../../tokens';
+
+/**
+ * A structural directive that provides a way to render UI elements based on whether a (set of) cookie(s) have been accepted by the user.
+ */
+@Directive({
+	selector: '[hasCookie]',
+	standalone: true,
+})
+export class NgxHasCookieDirective {
+	/**
+	 * The destroyed state of the directive
+	 */
+	private destroyed$: Subject<void>;
+
+	/**
+	 * The needed templateRefs
+	 */
+	private thenTemplateRef: TemplateRef<any> | null = null;
+	private thenViewRef: EmbeddedViewRef<any> | null = null;
+	private elseTemplateRef: TemplateRef<any> | null = null;
+	private elseViewRef: EmbeddedViewRef<any> | null = null;
+
+	/**
+	 * The ref of the component we wish to render as a fallback
+	 */
+	private componentRef: ComponentRef<NgxCookiesFallBackComponent>;
+
+	/**
+	 * The list of cookies we need to check
+	 */
+	private cookies: NgxHasCookieConfiguration[] = [];
+
+	/**
+	 * A cookie or list of cookies the item should have
+	 */
+	@Input() public set hasCookie(cookie: NgxHasCookieConfiguration | NgxHasCookieConfiguration[]) {
+		this.cookies = Array.isArray(cookie) ? cookie : [cookie];
+		this.updateView();
+	}
+
+	/**
+	 * The else template in case the cookie is not accepted
+	 */
+	@Input() public set hasCookieElse(ngTemplate: TemplateRef<any>) {
+		this.elseTemplateRef = ngTemplate;
+		this.elseViewRef = null;
+		this.updateView();
+	}
+
+	constructor(
+		templateRef: TemplateRef<any>,
+		private viewContainer: ViewContainerRef,
+		private readonly ngxCookieService: NgxCookieService,
+		private readonly cdRef: ChangeDetectorRef,
+		@Optional()
+		@Inject(NgxCookiesFallbackComponentToken)
+		private readonly component: Type<NgxCookiesFallBackComponent>
+	) {
+		this.thenTemplateRef = templateRef;
+	}
+
+	public ngOnDestroy(): void {
+		this.dispose();
+	}
+
+	/**
+	 * Updates the view and hides/renders the template as needed
+	 */
+	private updateView(): void {
+		// Iben: Dispose the current subscription
+		this.dispose();
+
+		// Iben: Create a new onDestroyed handler
+		this.destroyed$ = new Subject();
+
+		// Iben: Render the views based on the correct state
+		combineLatest(
+			// Iben Check for each cookie if it is accepted
+			this.cookies.map((cookie) => {
+				// Iben: If no specific services were provided, we can just check the category
+				if (!cookie.services) {
```

Comment:
A lot of if/else nesting here. Is this something we can split up into smaller (functional) methods to make it more comprehensible?